frontend/src/metabase/components/MetadataInfo/TableInfo/TableInfo.tsx

```diff
@@ -113,7 +114,9 @@ export function TableInfo({
       <MetadataContainer>
         <Fade visible={!hasFetchedMetadata}>
           <AbsoluteContainer>
-            <LoadingSpinner size={24} />
+            <Delayed delay={400}>
+              <LoadingSpinner size={24} />
+            </Delayed>
```

Comment:
The problem is that `hasFetchedMetadata` is always initially `false` because `isMissingFks` check is always `true` for tables without FKs.
If we could determine whether FKs need to be fetched for a given table, then we could fix this without additional components. I don't know if we have this information.
But the least we could do is to make sure that FKs are not refetched for a given table if they were already fetched once.

Reply:
Yes that was my initial reaction too but I'm afraid we don't have that information (ie. the state for "we haven't fetched fk's yet" and "we have fetched fk's but there are none" is the same), so I don't think we have this information.

The request to fetch Fk's is cached higher up the chain AFAICT so there are no superfluous requests going out.